### PR TITLE
for intercept call check for host match and add tests

### PR DESCRIPTION
**Pull request by weijjia**
1. For interceptor call, update the  check for backend call -- check if host matches 
2. add tests
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/RandalliLama%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23issuecomment-150435019%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-10-23T02%3A59%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%3Agun%3A%22%2C%20%22created_at%22%3A%20%222015-10-23T03%3A05%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-10-23T03%3A21%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22You%20might%20consider%20adding%20a%20comment%20that%20explains%20what%20that%20clause%20is%20about.%5Cr%5CnOtherwise%20I%27m%20signed%20off.%22%2C%20%22created_at%22%3A%20%222015-10-23T03%3A21%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%201c7ccc9c261e4703f9f685ea00711186a64de3a6%20lib/adal.js%2024%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23discussion_r42829632%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Under%20what%20circumstances%20do%20we%20end%20up%20here%3F%20%20Getting%20here%20would%20indicate%20that%20this%20is%20not%20an%20http%28s%29%20url.%20%20What%20kind%20of%20non-http%20redirect%20uri%27s%20get%20used%20in%20the%20context%20of%20adal.js%3F%22%2C%20%22created_at%22%3A%20%222015-10-23T03%3A07%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4307330%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RandalliLama%22%7D%7D%2C%20%7B%22body%22%3A%20%22In%20angular%20interceptor%20call%2C%20developer%20could%20do%20i.e%20%24http.get%28%27/someurl%27%29%2C%20this%20%27/someurl%27%20could%20be%20absolute%20url%20or%20relative%20url.%20If%20it%27s%20relative%2C%20then%20we%20need%20to%20treat%20it%20as%20from%20app%20backend.%20%22%2C%20%22created_at%22%3A%20%222015-10-23T03%3A20%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1166233%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/weijjia%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/adal.js%3AL776-789%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23issuecomment-150435019%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23issuecomment-150438083%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23issuecomment-150447337%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23discussion_r42829632%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23discussion_r42830080%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-js/pull/181%23issuecomment-150447552%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RandalliLama'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/181#issuecomment-150435019'>General Comment</a></b>
- <a href='https://github.com/RandalliLama'></a> :shipit:
- <a href='https://github.com/RandalliLama'></a> :shipit::gun:
- <a href='https://github.com/RandalliLama'></a> :shipit:
- <a href='https://github.com/RandalliLama'></a> You might consider adding a comment that explains what that clause is about.
Otherwise I'm signed off.
- [ ] <a href='#crh-comment-Pull 1c7ccc9c261e4703f9f685ea00711186a64de3a6 lib/adal.js 24'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/181#discussion_r42829632'>File: lib/adal.js:L776-789</a></b>
- <a href='https://github.com/RandalliLama'></a> Under what circumstances do we end up here?  Getting here would indicate that this is not an http(s) url.  What kind of non-http redirect uri's get used in the context of adal.js?
- <a href='https://github.com/weijjia'></a> In angular interceptor call, developer could do i.e $http.get('/someurl'), this '/someurl' could be absolute url or relative url. If it's relative, then we need to treat it as from app backend.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/181?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/181?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-js/pull/181?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/181'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-js/pull/181'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by RandalliLama** [comment]
:shipit:

**Comment by RandalliLama** [comment]
:shipit::gun:

File: lib/adal.js [hunk 1]
```diff
@@ -775,7 +776,13 @@ AuthenticationContext.prototype.getResourceForEndpoint = function (endpoint) {
...
     // App will use idtoken for calls to itself
-    if (endpoint.indexOf(this.config.redirectUri) > -1 || (endpoint.indexOf('http://') !== 0 && endpoint.indexOf('https://') !== 0)) {
+    // check if it's staring from http or https, needs to match with app host
+    if (endpoint.indexOf('http://') > -1 || endpoint.indexOf('https://') > -1) {
+        if (this._getHostFromUri(endpoint) === this._getHostFromUri(this.config.redirectUri)) {
+            return this.config.loginResource;
+        }
+    }
+    else {
         return this.config.loginResource;
```
Comment (RandalliLama) [inline lib/adal.js]: Under what circumstances do we end up here?  Getting here would indicate that this is not an http(s) url.  What kind of non-http redirect uri's get used in the context of adal.js?

Comment (weijjia) [inline lib/adal.js]: In angular interceptor call, developer could do i.e $http.get('/someurl'), this '/someurl' could be absolute url or relative url. If it's relative, then we need to treat it as from app backend.

**Comment by RandalliLama** [comment]
:shipit:

**Comment by RandalliLama** [comment]
You might consider adding a comment that explains what that clause is about.
Otherwise I'm signed off.

**Pull request merged**